// @ngx-firebase/client/fesm2015/ngx-firebase-client.js
import { __decorate, __param } from 'tslib';
import { Inject, ɵɵdefineInjectable, ɵɵinject, Injectable, NgModule } from '@angular/core';
import * as firebase from 'firebase/app';
import { apps, initializeApp } from 'firebase/app';
import 'firebase/firestore';
import 'firebase/auth';
import 'firebase/storage';
import { CommonModule } from '@angular/common';

let NgxFirebaseClientService = class NgxFirebaseClientService {
    constructor(firebaseConfig) {
        this.firebaseConfig = firebaseConfig;
    }
    get firebase() {
        if (!apps.length) {
            initializeApp(this.firebaseConfig);
        }
        return firebase;
    }
    auth() {
        return this.firebase.auth();
    }
    storage() {
        return this.firebase.storage();
    }
    firestore() {
        return this.firebase.firestore();
    }
};
NgxFirebaseClientService.ctorParameters = () => [
    { type: undefined, decorators: [{ type: Inject, args: ['firebaseConfig',] }] }
];
NgxFirebaseClientService.ɵprov = ɵɵdefineInjectable({ factory: function NgxFirebaseClientService_Factory() { return new NgxFirebaseClientService(ɵɵinject("firebaseConfig")); }, token: NgxFirebaseClientService, providedIn: "root" });
NgxFirebaseClientService = __decorate([
    Injectable({
        providedIn: 'root'
    }),
    __param(0, Inject('firebaseConfig'))
], NgxFirebaseClientService);

var NgxFirebaseClientModule_1;
let NgxFirebaseClientModule = NgxFirebaseClientModule_1 = class NgxFirebaseClientModule {
    static forRoot(firebaseConfig) {
        return {
            ngModule: NgxFirebaseClientModule_1,
            providers: [
                NgxFirebaseClientService,
                {
                    provide: 'firebaseConfig',
                    useValue: firebaseConfig
                }
            ]
        };
    }
};
NgxFirebaseClientModule = NgxFirebaseClientModule_1 = __decorate([
    NgModule({
        declarations: [],
        imports: [
            CommonModule
        ],
    })
], NgxFirebaseClientModule);

/*
 * Public API Surface of client
 */

/**
 * Generated bundle index. Do not edit.
 */

export { NgxFirebaseClientModule, NgxFirebaseClientService };
