// @ngx-firebase/client/esm5/lib/client.service.js
import { __decorate, __param } from "tslib";
import { Inject, Injectable } from '@angular/core';
import * as firebase from 'firebase/app';
// todo setup dynamic imports
import 'firebase/firestore';
import 'firebase/auth';
import 'firebase/storage';
import * as i0 from "@angular/core";
var NgxFirebaseClientService = /** @class */ (function () {
    function NgxFirebaseClientService(firebaseConfig) {
        this.firebaseConfig = firebaseConfig;
    }
    Object.defineProperty(NgxFirebaseClientService.prototype, "firebase", {
        get: function () {
            if (!firebase.apps.length) {
                firebase.initializeApp(this.firebaseConfig);
            }
            return firebase;
        },
        enumerable: true,
        configurable: true
    });
    NgxFirebaseClientService.prototype.auth = function () {
        return this.firebase.auth();
    };
    NgxFirebaseClientService.prototype.storage = function () {
        return this.firebase.storage();
    };
    NgxFirebaseClientService.prototype.firestore = function () {
        return this.firebase.firestore();
    };
    NgxFirebaseClientService.ctorParameters = function () { return [
        { type: undefined, decorators: [{ type: Inject, args: ['firebaseConfig',] }] }
    ]; };
    NgxFirebaseClientService.ɵprov = i0.ɵɵdefineInjectable({ factory: function NgxFirebaseClientService_Factory() { return new NgxFirebaseClientService(i0.ɵɵinject("firebaseConfig")); }, token: NgxFirebaseClientService, providedIn: "root" });
    NgxFirebaseClientService = __decorate([
        Injectable({
            providedIn: 'root'
        }),
        __param(0, Inject('firebaseConfig'))
    ], NgxFirebaseClientService);
    return NgxFirebaseClientService;
}());
export { NgxFirebaseClientService };
