// @ngx-firebase/client/esm5/lib/client.module.js
import { __decorate } from "tslib";
import { NgModule } from '@angular/core';
import { NgxFirebaseClientService } from './client.service';
import { CommonModule } from '@angular/common';
var NgxFirebaseClientModule = /** @class */ (function () {
    function NgxFirebaseClientModule() {
    }
    NgxFirebaseClientModule_1 = NgxFirebaseClientModule;
    NgxFirebaseClientModule.forRoot = function (firebaseConfig) {
        return {
            ngModule: NgxFirebaseClientModule_1,
            providers: [
                NgxFirebaseClientService,
                {
                    provide: 'firebaseConfig',
                    useValue: firebaseConfig
                }
            ]
        };
    };
    var NgxFirebaseClientModule_1;
    NgxFirebaseClientModule = NgxFirebaseClientModule_1 = __decorate([
        NgModule({
            declarations: [],
            imports: [
                CommonModule
            ],
        })
    ], NgxFirebaseClientModule);
    return NgxFirebaseClientModule;
}());
export { NgxFirebaseClientModule };
