// @ngx-firebase/client/esm2015/lib/client.service.js
import { __decorate, __param } from "tslib";
import { Inject, Injectable } from '@angular/core';
import * as firebase from 'firebase/app';
// todo setup dynamic imports
import 'firebase/firestore';
import 'firebase/auth';
import 'firebase/storage';
import * as i0 from "@angular/core";
let NgxFirebaseClientService = class NgxFirebaseClientService {
    constructor(firebaseConfig) {
        this.firebaseConfig = firebaseConfig;
    }
    get firebase() {
        if (!firebase.apps.length) {
            firebase.initializeApp(this.firebaseConfig);
        }
        return firebase;
    }
    auth() {
        return this.firebase.auth();
    }
    storage() {
        return this.firebase.storage();
    }
    firestore() {
        return this.firebase.firestore();
    }
};
NgxFirebaseClientService.ctorParameters = () => [
    { type: undefined, decorators: [{ type: Inject, args: ['firebaseConfig',] }] }
];
NgxFirebaseClientService.ɵprov = i0.ɵɵdefineInjectable({ factory: function NgxFirebaseClientService_Factory() { return new NgxFirebaseClientService(i0.ɵɵinject("firebaseConfig")); }, token: NgxFirebaseClientService, providedIn: "root" });
NgxFirebaseClientService = __decorate([
    Injectable({
        providedIn: 'root'
    }),
    __param(0, Inject('firebaseConfig'))
], NgxFirebaseClientService);
export { NgxFirebaseClientService };
